### ISSUE TITLE
All optional parameters not included

### ISSUE THREAD

**Issue by kimbell** [RSuter/NSwag#1232]
I've just updated to 11.16.0.0

The swagger spec has a method with a bunch of optional parameters. In the generated code, only the first optional parameter is included; the rest are ignored. 

This is an extract of the Swagger doc:

```json
"/journeys": {
      "get": {
        "tags": [
          "CheckIn"
        ],
        "summary": "Retrieve journeys",
        "operationId": "retrieveJourneys",
        "description": "Retrieve the list of journeys available for check-in corresponding to a confirmed booking.\nA journey will contain the information about flights and travelers associated to the booking.\nIn order to check-in the entire journey in one shot the operation to perform is POST /journey/{journeyId}/acceptance.\nIt is possible to retrieve or modify the information corresponding to any traveler on the flights associated to the journey, choosing among the available operations of the resource TravelerFlightInfo.\n",
        "parameters": [
          {
            "$ref": "#/parameters/lastName"
          },
          {
            "$ref": "#/parameters/optionalOrderId"
          },
          {
            "$ref": "#/parameters/eTicketNumber"
          },
          {
            "$ref": "#/parameters/optionalFrequentFlyerCardId"
          },
          {
            "$ref": "#/parameters/optionalDepartureDate"
          },
          {
            "$ref": "#/parameters/optionalOriginLocationCode"
          }
        ],
```
This is what is currently generated

```cs
public async System.Threading.Tasks.Task<JourneysListReply> RetrieveJourneysAsync(string lastName = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken))
        {
            var urlBuilder_ = new System.Text.StringBuilder();
            urlBuilder_.Append(BaseUrl != null ? BaseUrl.TrimEnd('/') : "").Append("/journeys?");
            if (lastName != null) urlBuilder_.Append("lastName=").Append(System.Uri.EscapeDataString(ConvertToString(lastName, System.Globalization.CultureInfo.InvariantCulture))).Append("&");
            urlBuilder_.Length--;
```

This is what was generated in a previous version
```cs
 public async System.Threading.Tasks.Task<JourneysListReply> RetrieveJourneysAsync(string lastName = null, string orderId = null, string eTicketNumber = null, string frequentFlyerCardId = null, System.DateTime? departureDate = null, string originLocationCode = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken))
        {
            var urlBuilder_ = new System.Text.StringBuilder();
            urlBuilder_.Append(BaseUrl != null ? BaseUrl.TrimEnd('/') : "").Append("/journeys?");
            if (lastName != null) urlBuilder_.Append("lastName=").Append(System.Uri.EscapeDataString(ConvertToString(lastName, System.Globalization.CultureInfo.InvariantCulture))).Append("&");
            if (orderId != null) urlBuilder_.Append("orderId=").Append(System.Uri.EscapeDataString(ConvertToString(orderId, System.Globalization.CultureInfo.InvariantCulture))).Append("&");
            if (eTicketNumber != null) urlBuilder_.Append("eTicketNumber=").Append(System.Uri.EscapeDataString(ConvertToString(eTicketNumber, System.Globalization.CultureInfo.InvariantCulture))).Append("&");
            if (frequentFlyerCardId != null) urlBuilder_.Append("frequentFlyerCardId=").Append(System.Uri.EscapeDataString(ConvertToString(frequentFlyerCardId, System.Globalization.CultureInfo.InvariantCulture))).Append("&");
            if (departureDate != null) urlBuilder_.Append("departureDate=").Append(System.Uri.EscapeDataString(departureDate.Value.ToString("s", System.Globalization.CultureInfo.InvariantCulture))).Append("&");
            if (originLocationCode != null) urlBuilder_.Append("originLocationCode=").Append(System.Uri.EscapeDataString(ConvertToString(originLocationCode, System.Globalization.CultureInfo.InvariantCulture))).Append("&");
            urlBuilder_.Length--;
```

**Comment by RSuter** [comment]
I suspect this is a regression of this PR: https://github.com/RSuter/NSwag/pull/1147

**Comment by RSuter** [comment]
Strange that this worked before :-)

**Comment by RSuter** [comment]
Fixed in v11.16.1

**Comment by kimbell** [comment]
Thanks for the quick fix :)
